File: src/main/java/com/dude/dms/ui/components/tags/TagContainer.java
package com.dude.dms.ui.components.tags;

import com.dude.dms.backend.data.tags.Tag;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;

public class TagContainer extends HorizontalLayout {

    public TagContainer(Iterable<Tag> tags) {
        getElement().getStyle().set("display", "inlineBlock");
        tags.forEach(tag -> add(new TagLabel(tag)));
    }
}

File: src/main/java/com/dude/dms/ui/components/tags/TagContainer.kt
package com.dude.dms.ui.components.tags

import com.dude.dms.backend.data.Tag
import com.vaadin.flow.component.orderedlayout.HorizontalLayout

class TagContainer(private var tags: Iterable<Tag>) : HorizontalLayout() {

    init {
        element.style["display"] = "inlineBlock"
        fill()
    }

    fun setTags(tags: Iterable<Tag>) {
        this.tags = tags
        fill()
    }

    fun fill() {
        removeAll()
        tags.forEach { add(TagLabel(it)) }
    }
}